src/app/icd/client/CheckInHandler.h

```diff
@@ -0,0 +1,94 @@
+/*
+ *
+ *    Copyright (c) 2023 Project CHIP Authors
+ *    All rights reserved.
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+/**
+ *    @file
+ *      This file defines objects for a CHIP CheckInMessage unsolicited
+ *      handler
+ *
+ */
+
+#pragma once
+
+#include <app/icd/client/CheckInDelegate.h>
+#include <app/icd/client/DefaultICDClientStorage.h>
+#include <lib/support/logging/CHIPLogging.h>
+#include <messaging/ExchangeContext.h>
+#include <messaging/ExchangeMgr.h>
+
+namespace chip {
+namespace app {
+class CheckInMessageHandler : public Messaging::ExchangeDelegate, public Messaging::UnsolicitedMessageHandler
```

Comment:
Why is the file called CheckInHandler but the class CheckInMessageHandler?

Reply:
Modified the class name to CheckInHandler.